packages/service-library/tests/rabbitmq/test_rabbitmq.py

```diff
@@ -400,9 +493,84 @@ async def test_rabbit_pub_sub_bind_and_unbind_topics(
     await _assert_message_received(mocked_message_parser, 0)
 
 
-@pytest.mark.no_cleanup_check_rabbitmq_server_has_no_errors()
+_TOPICS: Final[list[list[str] | None]] = [
+    None,
+    ["one"],
+    ["one", "two"],
+]
+
+
+@pytest.mark.parametrize("max_requeue_retry", [0, 1, 3, 10])
+@pytest.mark.parametrize("topics", _TOPICS)
+async def test__subscribe_to_failing_message_handler(
+    create_rabbitmq_client: Callable[[str], RabbitMQClient],
+    random_exchange_name: Callable[[], str],
+    random_rabbit_message: Callable[..., PytestRabbitMessage],
+    on_message_spy: mock.Mock,
+    max_requeue_retry: int,
+    topics: list[str] | None,
+):
+    async def _faulty_message_handler(message: Any) -> bool:
+        msg = f"Always fail. Received message {message}"
+        raise RuntimeError(msg)
+
+    topics_multiplier = await _setup_publisher_and_subscriber(
+        create_rabbitmq_client,
+        random_exchange_name,
+        random_rabbit_message,
+        max_requeue_retry,
+        topics,
+        _faulty_message_handler,
+    )
+
+    expected_results = (max_requeue_retry + 1) * topics_multiplier
+    await _assert_wait_for_messages(on_message_spy, expected_results)
+
+    report = _get_spy_report(on_message_spy)
+    routing_keys: list[str] = [""] if topics is None else topics
+    assert report == {k: set(range(max_requeue_retry + 1)) for k in routing_keys}
+
+
+@pytest.mark.parametrize("topics", _TOPICS)
+async def test__subscribe_no_dead_letter_exchange_messages(
+    create_rabbitmq_client: Callable[[str], RabbitMQClient],
+    random_exchange_name: Callable[[], str],
+    random_rabbit_message: Callable[..., PytestRabbitMessage],
+    on_message_spy: mock.Mock,
+    topics: list[str] | None,
+):
+    message_failed: dict[str, bool] = {}
+
+    async def _fail_once_then_succeed(message: Any) -> bool:
+        if message not in message_failed:
+            message_failed[message] = False
+        if not message_failed[message]:
+            message_failed[message] = True
+            return False
+        return True
+
+    topics_multiplier = await _setup_publisher_and_subscriber(
+        create_rabbitmq_client,
+        random_exchange_name,
+        random_rabbit_message,
+        DEFAULT_UNEXPECTED_ERROR_MAX_ATTEMPTS,
+        topics,
+        _fail_once_then_succeed,
+    )
+
+    expected_results = 2 * topics_multiplier
+    await _assert_wait_for_messages(on_message_spy, expected_results)
+
+    report = _get_spy_report(on_message_spy)
+    routing_keys: list[str] = [""] if topics is None else topics
+    assert report == {k: set(range(1)) for k in routing_keys}
+
+    # ensure no headers in received messages, meaning the handler did not fail
+    for entry in on_message_spy.call_args_list:
+        assert entry.args[0].headers == {}
+
+
 async def test_rabbit_adding_topics_to_a_fanout_exchange(
-    cleanup_check_rabbitmq_server_has_no_errors: None,
```

Comment:
if you do not put that fixture then it will not check the rabbit logs for error. Please add it back. or else explain why it goes away.

Reply:
If I read the below lines write, this is automatically used. Since it's in the `conftest.py` module. I would prefer to have it explicitly used. I have to use the decorator `@pytest.mark.no_cleanup_check_rabbitmq_server_has_no_errors()` to disable it's usage, like I did inside the `packages/service-library/tests/rabbitmq/test_rabbitmq_utils.py` which was failing to run locally

https://github.com/ITISFoundation/osparc-simcore/blob/567c0a37c99ffdbd0473c6341cb23176db806c8a/packages/service-library/tests/rabbitmq/conftest.py#L10-L11

Reply:
already discussed about this one